backend/Application/Interop/InteropBinding.cs

```diff
@@ -0,0 +1,131 @@
+    ///
+    /// Receive parameters are those given to a contract entrypoint on a update call.
+    /// </summary>
+    /// <param name="schema">Module schema in hexadecimal</param>
+    /// <param name="contractName">Contract name</param>
+    /// <param name="entrypoint">Entrypoint of contract</param>
+    /// <param name="value">Receive parameters in hexadecimal</param>
+    /// <param name="schemaVersion">Optional schema version if present from module</param> 
+    /// <returns>Receive parameters in a human interpretable form</returns>
+    internal static InteropResult GetReceiveContractParameter(string schema, string contractName, string entrypoint, string value, FFIOption schemaVersion)
+    {
+        var result = IntPtr.Zero;
+        try
+        {
+            var schemaDisplay = get_receive_contract_parameter(schema, schemaVersion, contractName, entrypoint, value, ref result);
+            var ptrToStringAnsi = Marshal.PtrToStringAnsi(result);
+            return new InteropResult(ptrToStringAnsi, schemaDisplay);
+        }
+        finally
+        {
+            Free(result);    
+        }
+    }
+    
+    /// <summary>
+    /// Get contract event in a human interpretable form.
+    /// </summary>
+    /// <param name="schema">Module schema in hexadecimal</param>
+    /// <param name="contractName">Contract name</param>
+    /// <param name="value">Contract event in hexadecimal</param>
+    /// <param name="schemaVersion">Optional schema version if present from module</param>
+    /// <returns>Contract event in a human interpretable form</returns>
+    internal static InteropResult GetEventContract(string schema, string contractName, string value, FFIOption schemaVersion)
+    {
+        var result = IntPtr.Zero;
+        try
+        {
+            var schemaDisplay = get_event_contract(schema, schemaVersion, contractName, value, ref result);
+            var ptrToStringAnsi = Marshal.PtrToStringAnsi(result);
+            return new InteropResult(ptrToStringAnsi, schemaDisplay);
+        }
+        finally
+        {
+            Free(result);   
+        }
+    }
+    
+    /// <summary>
+    /// Wrap a interop result. If the interop call <see cref="Succeeded"/> the <see cref="Message"/> contains the result.
+    /// If it failed the <see cref="Message"/> will contain the error message. 
+    /// </summary>
+    /// <param name="Message">Message from interop operation.</param>
+    /// <param name="Succeeded">If the call succeeded.</param>
+    internal readonly record struct InteropResult(string? Message, bool Succeeded);
+    
+    private static void Free(IntPtr ptr)
+    {
+        if (ptr != IntPtr.Zero)
+        {
+            Marshal.FreeHGlobal(ptr);
+        }
+    }
+    
+    /// <summary>
+    /// A C# layout which compiled to a C interpretable structure. This is used as an optional parameter.
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    public struct FFIOption
```

Comment:
Maybe let name indicate that it's an option of byte?

Reply:
I’m only using a byte for boolean because it can’t be directly mapped and a byte for the enum value. I rather want to call it something like `FFIEnumOption` if I should update it, since this maybe represents better usages of it :-)

Reply:
I'm referring to the byte value `t`, not the `is_some` boolean.

Calling it `FFIEnumOption` would only be accurate for enums that happen to turn into a byte (like schema version), so I think that would be misleading.